pkg/helm/helm.go

```diff
@@ -239,32 +262,95 @@ type TemplateOptions struct {
 }
 
 func (c *Client) Template(ctx context.Context, chart string, opts TemplateOptions) ([]byte, error) {
-	if opts.Name == "" {
-		opts.GenerateName = true
+	// NOTE: Unlike other methods, Template calls into helm directly. This is
+	// to minimize any potential overhead from go/helm/cobra's start up time
+	// and allow us to be much more aggressive with writing tests through
+	// Template.
+	// TODO: Support IsUpgrade and the like and find a nice way to inject a
+	// fake KubeClient.
+	client := action.NewInstall(c.config)
+
+	client.APIVersions = chartutil.DefaultVersionSet
+	client.ClientOnly = true
+	client.DryRun = true
+	client.GenerateName = opts.GenerateName
+	client.Namespace = opts.Namespace
+	client.ReleaseName = opts.Name
+	client.Replace = true // Skips "name" checks.
```

Comment:
are we sure we want this default value in production?

Reply:
This is borrowed directly from helm itself. `helm template` needs to run in an isolated environment. Rather than having that logic baked in, they opted to use this as a work around. I'll add a back link and a note saying as much. I could have sworn that I had a few links peppered in there already but looks pretty scant.